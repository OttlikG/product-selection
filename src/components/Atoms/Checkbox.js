
import React from 'react'

export default function Checkbox (props) {
  const { label, isSelected, onChange } = props

  return (
    <div className="checkbox">
      <label>
        <input type="checkbox" checked={isSelected} onChange={onChange} /> {label}
      </label>
    </div>
  )
}
