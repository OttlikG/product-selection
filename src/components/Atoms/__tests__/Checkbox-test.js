
import React from 'react'
import Checkbox from 'components/Atoms/Checkbox'

describe('Checkbox', () => {

  let props
  let component

  beforeEach(() => {
    props = {
      label: 'test label',
      isSelected: true,
      onChange: jest.fn(),
    }

    component = render(<Checkbox {...props} />)
  })

  it('should set props on input', () => {
    const input = component.find('input')

    expect(input.props.checked).toBe(true)
    expect(input.props.onChange).toBeDefined()
  })

  it('should load label text', () => {
    const label = component.text('label')
    expect(label).toBe(' test label')
  })

  it('should call on change', () => {
    const input = component.find('input')
    input.props.onChange()
    expect(props.onChange).toHaveBeenCalled()
  })
})
