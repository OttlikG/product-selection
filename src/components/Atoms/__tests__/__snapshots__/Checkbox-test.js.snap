// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should call on change 1`] = `
<div
  className="checkbox"
>
  <label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
     
    test label
  </label>
</div>
`;

exports[`Checkbox should load label text 1`] = `
<div
  className="checkbox"
>
  <label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
     
    test label
  </label>
</div>
`;

exports[`Checkbox should set props on input 1`] = `
<div
  className="checkbox"
>
  <label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
     
    test label
  </label>
</div>
`;
